tests/performance/timing_step_68.cc

```diff
@@ -0,0 +1,510 @@
+      MPI_COMM_WORLD);
+
+    GridGenerator::hyper_cube(particle_triangulation, 0, 1);
+    particle_triangulation.refine_global(par.particle_insertion_refinement);
+
+    const auto my_bounding_box = GridTools::compute_mesh_predicate_bounding_box(
+      background_triangulation, IteratorFilters::LocallyOwnedCell());
+    const auto global_bounding_boxes =
+      Utilities::MPI::all_gather(MPI_COMM_WORLD, my_bounding_box);
+
+    std::vector<std::vector<double>> properties(
+      particle_triangulation.n_locally_owned_active_cells(),
+      std::vector<double>(dim + 1, 0.));
+
+    Particles::Generators::quadrature_points(particle_triangulation,
+                                             QMidpoint<dim>(),
+                                             global_bounding_boxes,
+                                             particle_handler,
+                                             mapping,
+                                             properties);
+
+    pcout << "Number of particles inserted: "
+          << particle_handler.n_global_particles() << std::endl;
+  }
+
+
+
+  template <int dim>
+  void
+  ParticleTracking<dim>::setup_background_dofs()
+  {
+    fluid_dh.distribute_dofs(fluid_fe);
+    const IndexSet locally_owned_dofs = fluid_dh.locally_owned_dofs();
+    const IndexSet locally_relevant_dofs =
+      DoFTools::extract_locally_relevant_dofs(fluid_dh);
+
+    velocity_field.reinit(locally_owned_dofs,
+                          locally_relevant_dofs,
+                          mpi_communicator);
+  }
+
+
+
+  template <int dim>
+  void
+  ParticleTracking<dim>::interpolate_function_to_field()
+  {
+    velocity_field.zero_out_ghost_values();
+    VectorTools::interpolate(mapping, fluid_dh, velocity, velocity_field);
+    velocity_field.update_ghost_values();
+  }
+
+
+  template <int dim>
+  void
+  ParticleTracking<dim>::euler_step_interpolated(const double dt)
+  {
+    Vector<double> local_dof_values(fluid_fe.dofs_per_cell);
+
+    auto particle = particle_handler.begin();
+    while (particle != particle_handler.end())
+      {
+        const auto cell = particle->get_surrounding_cell();
+        const auto dh_cell =
+          typename DoFHandler<dim>::cell_iterator(*cell, &fluid_dh);
+
+        dh_cell->get_dof_values(velocity_field, local_dof_values);
+
+        const auto pic = particle_handler.particles_in_cell(cell);
+        Assert(pic.begin() == particle, ExcInternalError());
+        for (auto &p : pic)
+          {
+            const Point<dim> reference_location = p.get_reference_location();
+            Tensor<1, dim>   particle_velocity;
+            for (unsigned int j = 0; j < fluid_fe.dofs_per_cell; ++j)
+              {
+                const auto comp_j = fluid_fe.system_to_component_index(j);
+
+                particle_velocity[comp_j.first] +=
+                  fluid_fe.shape_value(j, reference_location) *
+                  local_dof_values[j];
+              }
```

Comment:
Can we use `FEPointEvaluation` for this evaluation? That class was written for this use case, and it is more performance-oriented than the virtual functions in `FiniteElement::shape_value`? (I should add that the main performance bottleneck is not the virtual function, but the fact that we need to go to all 1D polynomials of the tensor product polynomial.) If you want, I can add that as a follow-up PR in the coming days, we can get this first version in quickly in my opinion.

Reply:
@kronbichler absolutely. What do you prefer? We merge this one then you make a PR to fix this + step-68 with the correct FEEvaluation?

That might be the best way to go?

Reply:
Yes, let's merge this test first.